Plugins/NativeCamera/Source/NativeCamera.cpp

```diff
@@ -0,0 +1,166 @@
+#include "NativeCamera.h"
+
+#include <Babylon/JsRuntime.h>
+#include <GraphicsImpl.h>
+#include "NativeCamera.h"
+#include <vector>
+
+namespace Babylon::Plugins::Internal
+{
+    class NativeCamera : public Napi::ObjectWrap<NativeCamera>
+    {
+        static constexpr auto JS_NAVIGATOR_NAME = "navigator";
+        static constexpr auto JS_CLASS_NAME = "_NativeCamera";
+        static constexpr auto JS_NATIVECAMERA_CONSTRUCTOR_NAME = "NativeCamera";
+        static constexpr auto JS_NATIVECAMERA_DATA = "_nativeCameraData";
+
+    public:
+
+        static void Initialize(Napi::Env env)
+        {
+            Napi::HandleScope scope{ env };
+
+            Napi::Function func = NativeCamera::DefineClass(
+                env,
+                JS_CLASS_NAME,
+                {
+                    NativeCamera::InstanceMethod("createVideo", &NativeCamera::CreateVideo),
+                    NativeCamera::InstanceMethod("updateVideoTexture", &NativeCamera::UpdateVideoTexture),
+                });
+
+            JsRuntime::NativeObject::GetFromJavaScript(env).Set(JS_NATIVECAMERA_CONSTRUCTOR_NAME, func);
+
+            // create or get global navigator object
+            Napi::Object global = env.Global();
+            Napi::Object navigator;
+            if (global.Has(JS_NAVIGATOR_NAME))
+            {
+                navigator = global.Get(JS_NAVIGATOR_NAME).As<Napi::Object>();
+            }
+            else
+            {
+                navigator = Napi::Object::New(env);
+                global.Set(JS_NAVIGATOR_NAME, navigator);
+            }
+
+            // append media devices to navigator
+            Napi::Object mediaDevices = Napi::Object::New(env);
+            mediaDevices.Set("getUserMedia", Napi::Function::New(env, [](const Napi::CallbackInfo& info) {
+                auto env = info.Env();
+                auto deferred{ Napi::Promise::Deferred::New(env) };
+                auto promise{ deferred.Promise() };
+               
+                auto& jsRuntime{ JsRuntime::GetFromJavaScript(env) };
+                    jsRuntime.Dispatch([deferred{ std::move(deferred) }](Napi::Env env) {
+                        deferred.Resolve(env.Null());
+                    });
+                
+                return promise;
+                }));
+            navigator.Set("mediaDevices", mediaDevices);
+        }
+
+        NativeCamera(const Napi::CallbackInfo& info)
+            : Napi::ObjectWrap<NativeCamera>{ info }
+        {
+        }
+
+        ~NativeCamera()
+        {
+        }
+
+    private:
+        Napi::Value CreateVideo(const Napi::CallbackInfo& info)
+        {
+            const auto env = info.Env();
+            Napi::Object video = Napi::Object::New(env);
```

Comment:
nit: Would it make sense to split this out into an `ObjectWrap`, similar to what we do in a few cases in NativeXr? Since this thing has methods, it just seems a little hefty to be constructing programmatically, though it probably ultimately resolves to the same thing and is thus more of a stylistic choice than anything else.